tools/IoTEdgeDevOps/VstsPipelineSync/VstsBuildBatchUpdate.cs

```diff
@@ -0,0 +1,122 @@
+namespace VstsPipelineSync
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Data.SqlClient;
+    using System.Linq;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using DevOpsLib;
+    using DevOpsLib.VstsModels;
+
+    public class VstsBuildBatchUpdate
+    {
+        readonly DevOpsAccessSetting devOpsAccessSetting;
+        readonly string dbConnectionString;
+
+        public VstsBuildBatchUpdate(DevOpsAccessSetting devOpsAccessSetting, string dbConnectionString)
+        {
+            ValidationUtil.ThrowIfNull(devOpsAccessSetting, nameof(devOpsAccessSetting));
+
+            this.devOpsAccessSetting = devOpsAccessSetting;
+            this.dbConnectionString = dbConnectionString;
+        }
+
+        public async Task RunAsync(TimeSpan waitPeriodAfterEachUpdate, CancellationToken ct)
+        {
+            var buildManagement = new BuildManagement(devOpsAccessSetting);
+            var lastUpdatePerBranchPerDefinition = new Dictionary<string, Dictionary<BuildDefinitionId, DateTime>>();
```

Comment:
nit: this could be a member variable and then it we wouldn't need to define it inside the func and pass it as an arg to another func.